engine/src/main/java/org/terasology/math/AABB.java

```diff
@@ -146,15 +144,56 @@ public AABB move(Vector3f offset) {
     }
 
     public AABB transform(Quat4f rotation, Vector3f offset, float scale) {
-        Transform transform = new Transform(new Matrix4f(VecMath.to(rotation), VecMath.to(offset), scale));
+        Transform transform = new Transform(offset, rotation, scale);
         return transform(transform);
```

Comment:
This is a bit ugly. I do not understand, by looking at this code, what the transform does to itself.

Reply:
Renaming the `transform` method to `transformWith` might help. Arguably, moving this method into the `Transform` class itself (`AABB Transform.transformAabb(AABB aabb)`) would be even better, but that would bring about even more changes not directly related to the physics engine. I think this change would be better off as another PR once this is merged, but if you don't think so I can do it here.

Reply:
Just for clarity, the transform does not do anything to itself. The `this` AABB is transformed (rotated, translated and scaled) according to the `transform` object and the transformed AABB is returned.

Reply:
I know what you mean. It's just that this code seems to say that a transform is being transformed. When instead it is being passed through a chain of "transform" methods that (I'm guessing) are all needed at one point or another, down to the one actually doing the work. 

Hmmm. I can't think of a way to write it differently though. So, let's leave it as it is for now.

Reply:
I agree that it can be made more readable. Moving the method into the `Transform` class as I mentioned before would, in my opinion, make things much better, but this method must have been used in other places outside the physics engine, making a refactor out of scope for this PR.

Reply:
Similar to how the `Transform.transform` method (which, I think should be renamed to `transformPosition`) is:
```
public AABB transformAabb(AABB aabb)
```

Reply:
Now that I think about this with a fresh mind, I have to agree with your assessment. It makes sense to rename this method to `transformWith` or `transformTo` or similar (or something entirely different), but I think this change (and any further deliberations) will be better suited to the subsequent Transform improvement PR.

Reply:
Ok.

Reply:
Just write it down somewhere!